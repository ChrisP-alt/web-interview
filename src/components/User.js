import React from 'react'
import PropTypes from 'prop-types'

import styles from '../styles/components/user.module.scss'

const User = ({ user }) => {
  if (!user) {
    return null
  }
  return (
    <div className={styles.wrapper}>
      <div className={styles.userDetails}>
        <img src={user.avatar} alt="User Avatar" />
        <span>{`${user.firstName} ${user.lastName}`}</span>
      </div>
      <a href="#todo" className={styles.change}>
        Change
      </a>
    </div>
  )
}

User.propTypes = {
  user: PropTypes.shape({
    id: PropTypes.number.isRequired,
    firstName: PropTypes.string.isRequired,
    lastName: PropTypes.string.isRequired,
    avatar: PropTypes.string.isRequired,
    dateOfBirth: PropTypes.string.isRequired,
  }),
}

export default User
