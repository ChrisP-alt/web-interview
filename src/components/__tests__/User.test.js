import React from 'react'
import { shallow } from 'enzyme'

import User from '../User'
import mockUser from '../../__fixtures__/user'

describe('User component', () => {
  it('renders correctly with a user', () => {
    const wrapper = shallow(<User user={mockUser} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders null without a user', () => {
    const wrapper = shallow(<User />)
    expect(wrapper.getElement()).toBe(null)
  })
})
