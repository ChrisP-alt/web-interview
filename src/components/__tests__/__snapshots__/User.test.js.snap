// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User component renders correctly with a user 1`] = `
<div
  className="wrapper"
>
  <div
    className="userDetails"
  >
    <img
      alt="User Avatar"
      src="https://fake.image.url"
    />
    <span>
      Jane Doe
    </span>
  </div>
  <a
    className="change"
    href="#todo"
  >
    Change
  </a>
</div>
`;
